File: 对齐线/Qt/button.cpp
#include "button.h"
#include "window.h"

Button::Button(QString text, QWidget *parent):QPushButton(text, parent)
{

}

Button::~Button() {

}

void Button::mousePressEvent(QMouseEvent *event) {
    QPushButton::mousePressEvent(event);
    this->startX = event->x();
    this->startY = event->y();
}

void Button::mouseMoveEvent(QMouseEvent *event) {
    QPushButton::mouseMoveEvent(event);
    int disX = event->x() - this->startX;
    int disY = event->y() - this->startY;
    this->move(this->x()+disX, this->y()+disY);

    Window *win = (Window*) this->parent();
    win->checkPos(this);
}

void Button::mouseReleaseEvent(QMouseEvent *event) {
    QPushButton::mouseReleaseEvent(event);
}

File: 对齐线/PyQt/button.py
from PyQt5.QtWidgets import QPushButton


class Button(QPushButton):
    def __init__(self, text, parent):
        super(Button, self).__init__(text, parent)
        self.start_x = None
        self.start_y = None
        
    def mousePressEvent(self, event):
        super(Button, self).mousePressEvent(event)
        self.start_x = event.x()
        self.start_y = event.y()
        
    def mouseMoveEvent(self, event):
        super(Button, self).mouseMoveEvent(event)
        dis_x = event.x() - self.start_x
        dis_y = event.y() - self.start_y
        self.move(self.x()+dis_x, self.y()+dis_y)

        self.parent().check_pos(self)

    def mouseReleaseEvent(self, event):
        super(Button, self).mouseReleaseEvent(event)
        self.parent().is_vertical_center_line_shown = False
        self.parent().is_horizontal_center_line_shown = False
        self.parent().is_btn_left_line_shown = False
        self.parent().is_btn_top_line_shown = False
        self.parent().update()
